teamcity: failed tests on master: acceptance/TestDockerMigration_7429

The following tests appear to have failed:

[#182457](https://teamcity.cockroachdb.com/viewLog.html?buildId=182457):

```
--- FAIL: acceptance/TestDockerMigration_7429 (6.190s)
reference_test.go:41: non-zero exit code: 1

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/cli/start.go
+++ b/pkg/cli/start.go
@@ -617,13 +617,14 @@ func doShutdown(ctx context.Context, c serverpb.AdminClient, onModes []int32) er
 		// not already trying to do that, and if the initial connection attempt
 		// (without shutdown) has succeeded.
 		tryHard := i > 0 && len(onModes) > 0
-		var hasResp bool
 		for {
 			if _, err := stream.Recv(); err != nil {
-				if hasResp && grpcutil.IsClosedConnection(err) {
-					// We're trying to shut down, and we already got a response
-					// and now the connection is closed. This means we shut
-					// down successfully.
+				if grpcutil.IsClosedConnection(err) {
+					// We're trying to shut down, we successfully sent a drain request,
+					// and now the connection is closed. This most likely means that we
+					// shut down successfully. Note that sometimes the stream can be shut
+					// down even before a DrainResponse gets sent back to us, so we don't
+					// require a response on the stream (see #14184).
 					return nil
 				}
 				if tryHard {
@@ -640,7 +641,6 @@ func doShutdown(ctx context.Context, c serverpb.AdminClient, onModes []int32) er
 				// Our liveness probe succeeded.
 				break
 			}
-			hasResp = true
 		}
 	}
 	return nil

--- a/pkg/util/grpcutil/grpc_util.go
+++ b/pkg/util/grpcutil/grpc_util.go
@@ -38,6 +38,7 @@ func IsClosedConnection(err error) bool {
 		grpc.Code(err) == codes.Unavailable ||
 		grpc.ErrorDesc(err) == grpc.ErrClientConnClosing.Error() ||
 		strings.Contains(err.Error(), "is closing") ||
+		strings.Contains(err.Error(), "use of closed network connection") ||
 		strings.Contains(err.Error(), "node unavailable") {
 		return true
 	}

